### PR TITLE
Performance improvement

### PR DESCRIPTION
**Pull request by Raycoms**
Only tick on tickphase end should make us perform way better.

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/colony/buildings/BuildingHome.java [hunk 1]
```diff
@@ -199,11 +199,6 @@ public void removeCitizen(@NotNull final CitizenData citizen)
     @Override
     public void onWorldTick(@NotNull final TickEvent.WorldTickEvent event)
     {
-        if (event.phase != TickEvent.Phase.END)
```
Comment (marchermans) [inline src/main/java/com/minecolonies/coremod/colony/buildings/BuildingHome.java:202]: Why is this removed?

**Review by marchermans — CHANGES_REQUESTED** [review]

File: src/main/java/com/minecolonies/coremod/colony/buildings/BuildingHome.java (hunk 1 above)
Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/colony/buildings/BuildingHome.java:202]: because the building get called over the colonyManager don't they? So we don't have to check it twice.

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 78 issues
* ![CRITICAL][CRITICAL] 2 critical
* ![MAJOR][MAJOR] 60 major
* ![MINOR][MINOR] 14 minor
* ![INFO][INFO] 2 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [ColonyManager.java#L956](https://github.com/Minecolonies/minecolonies/blob/2ba1c6df78197e95bf7aba4e83354eab70c2e403/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L956): Refactor this code to not nest more than 5 if/for/while/switch/try statements. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS134)
1. ![CRITICAL][CRITICAL] [AbstractBuildingWorker.java#L88](https://github.com/Minecolonies/minecolonies/blob/2ba1c6df78197e95bf7aba4e83354eab70c2e403/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingWorker.java#L88): Remove this call from a constructor to the overridable "getBuildingLevel" method. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1699)
1. ![MAJOR][MAJOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/2ba1c6df78197e95bf7aba4e83354eab70c2e403/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/2ba1c6df78197e95bf7aba4e83354eab70c2e403/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [render#](https://github.com/Minecolonies/minecolonies/blob/2ba1c6df78197e95bf7aba4e83354eab70c2e403/src/main/java/com/minecolonies/coremod/client/render#): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [colony#](https://github.com/Minecolonies/minecolonies/blob/2ba1c6df78197e95bf7aba4e83354eab70c2e403/src/main/java/com/minecolonies/coremod/colony#): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [Colony.java#L59](https://github.com/Minecolonies/minecolonies/blob/2ba1c6df78197e95bf7aba4e83354eab70c2e403/src/main/java/com/minecolonies/coremod/colony/Colony.java#L59): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 38 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L59](https://github.com/Minecolonies/minecolonies/blob/2ba1c6df78197e95bf7aba4e83354eab70c2e403/src/main/java/com/minecolonies/coremod/colony/Colony.java#L59): class "Colony" has 65 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [ColonyManager.java#L55](https://github.com/Minecolonies/minecolonies/blob/2ba1c6df78197e95bf7aba4e83354eab70c2e403/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L55): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 38 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [ColonyManager.java#L55](https://github.com/Minecolonies/minecolonies/blob/2ba1c6df78197e95bf7aba4e83354eab70c2e403/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L55): class "ColonyManager" has 57 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingWorker.java [hunk 2]
```diff
@@ -463,11 +463,6 @@ public void onWorldTick(@NotNull final TickEvent.WorldTickEvent event)
     {
         super.onWorldTick(event);
 
-        if (event.phase != TickEvent.Phase.END)
-        {
-            return;
-        }
-
         // If we have no active worker, grab one from the Colony
         // TODO Maybe the Colony should assign jobs out, instead?
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingWorker.java:467]: INFO Complete the task associated to this TODO comment. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 78 issues
* ![CRITICAL][CRITICAL] 2 critical
* ![MAJOR][MAJOR] 60 major
* ![MINOR][MINOR] 14 minor
* ![INFO][INFO] 2 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [ColonyManager.java#L956](https://github.com/Minecolonies/minecolonies/blob/2ba1c6df78197e95bf7aba4e83354eab70c2e403/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L956): Refactor this code to not nest more than 5 if/for/while/switch/try statements. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS134)
1. ![CRITICAL][CRITICAL] [AbstractBuildingWorker.java#L88](https://github.com/Minecolonies/minecolonies/blob/2ba1c6df78197e95bf7aba4e83354eab70c2e403/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingWorker.java#L88): Remove this call from a constructor to the overridable "getBuildingLevel" method. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1699)
1. ![MAJOR][MAJOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/2ba1c6df78197e95bf7aba4e83354eab70c2e403/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/2ba1c6df78197e95bf7aba4e83354eab70c2e403/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [render#](https://github.com/Minecolonies/minecolonies/blob/2ba1c6df78197e95bf7aba4e83354eab70c2e403/src/main/java/com/minecolonies/coremod/client/render#): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [colony#](https://github.com/Minecolonies/minecolonies/blob/2ba1c6df78197e95bf7aba4e83354eab70c2e403/src/main/java/com/minecolonies/coremod/colony#): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [Colony.java#L59](https://github.com/Minecolonies/minecolonies/blob/2ba1c6df78197e95bf7aba4e83354eab70c2e403/src/main/java/com/minecolonies/coremod/colony/Colony.java#L59): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 38 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L59](https://github.com/Minecolonies/minecolonies/blob/2ba1c6df78197e95bf7aba4e83354eab70c2e403/src/main/java/com/minecolonies/coremod/colony/Colony.java#L59): class "Colony" has 65 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [ColonyManager.java#L55](https://github.com/Minecolonies/minecolonies/blob/2ba1c6df78197e95bf7aba4e83354eab70c2e403/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L55): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 38 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [ColonyManager.java#L55](https://github.com/Minecolonies/minecolonies/blob/2ba1c6df78197e95bf7aba4e83354eab70c2e403/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L55): class "ColonyManager" has 57 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/main/java/com/minecolonies/coremod/colony/buildings/BuildingHome.java (hunk 1 above)
Comment (marchermans) [reply src/main/java/com/minecolonies/coremod/colony/buildings/BuildingHome.java:202]: Put it in just to be sure. When we rewrite we will face this anyway. So we can do a double check then.

**Review by marchermans — APPROVED** [review]
